src/main/kotlin/oop/proxy/VirtualProxy.kt

```diff
@@ -0,0 +1,28 @@
+package oop.proxy
+
+interface Screen {
+  fun show()
+}
+
+class ScreenVirtualProxy(val createRealScreen: () -> Screen) : Screen {
```

Comment:
I think we could make use of kotlin's `lazy` delegation in this case, I do not really know if you are already instanciating the real object if you use the constructor as a high order function 🤔